src/Resolver.php

```diff
@@ -4,38 +4,36 @@
 
 namespace OpenEuropa\ComposerDependentPatches;
 
+use Composer\Plugin\PluginInterface;
 use cweagans\Composer\Resolver as ComposerPatchesResolver;
 use OpenEuropa\ComposerDependentPatches\Resolver\Dependencies;
 use OpenEuropa\ComposerDependentPatches\Resolver\RootComposer;
 
 class Resolver extends ComposerPatchesResolver
 {
     /**
-     * {@inheritdoc}
+     * @var PluginInterface
      */
-    protected function getPatchResolvers(): array
+    protected PluginInterface $plugin;
+
+    /**
+     * Constructor.
+     */
+    public function __construct($composer, $io, $config, PluginInterface $plugin)
```

Comment:
I think it would be cleaner to match parent's signature:
see: `vendor/cweagans/composer-patches/src/Resolver.php`

```suggestion
    public function __construct(Composer $composer, IOInterface $io, array $disabledResolvers, PluginInterface $plugin)
```